rust/aptos-indexer-transaction-stream/src/transaction_stream.rs

```diff
@@ -296,14 +299,7 @@ pub async fn get_chain_id(
     }
 }
 
-/// Gets a batch of transactions from the stream. Batch size is set in the grpc server.
-/// The number of batches depends on our config
-/// There could be several special scenarios:
-/// 1. If we lose the connection, we will try reconnecting X times within Y seconds before crashing.
-/// 2. If we specified an end version and we hit that, we will stop fetching, but we will make sure that
-///    all existing transactions are processed
-pub async fn create_fetcher_loop(
-    txn_sender: AsyncSender<TransactionsPBResponse>,
+pub struct TransactionStream {
```

Comment:
Doc comments at the struct and field level would be good since this is a public struct. Just for the more potentially confusing fields.